hpx: TeamPolicy routine updates for large number of hw threads

ndellingwood opened the pull request:
This PR updates the setup() and execute_range() member routines of the ParallelFor specialization for HPX TeamPolicy to constrain hpx_thread_buffer size when running on hardware with large numbers of threads

These changes were made after observing std::bad_alloc failures in the `hpx.large_team_scratch_size` when testing on Intel SPX arch (192 hw threads) due to request for too much memory

Changes to resolve the failure were suggested by @crtrott

ndellingwood committed:
hpx: TeamPolicy routine updates for large number of hw threads

This PR updates the setup() and execute_range() member routines of the ParallelFor specialization for HPX TeamPolicy to constrain hpx_thread_buffer size when running on hardware with large numbers of threads

These changes were made after observing std::bad_alloc failures in the `hpx.large_team_scratch_size` when testing on Intel SPX arch (192 hw threads) due to request for too much memory

Changes to resolve the failure were suggested by @crtrott

Co-authored-by: Christian Trott <crtrott@sandia.gov>
Signed-off-by: Nathan Ellingwood <ndellin@sandia.gov>

ndellingwood committed:
apply clang-format

Signed-off-by: Nathan Ellingwood <ndellin@sandia.gov>

ndellingwood commented:
The test failure was observed while preparing to move the HPX CI testing to SNL resources, related to #8145

crtrott reviewed (dismissed):
What about ParallelReduce? Does it have the same issue?

ndellingwood commented:
> What about ParallelReduce? Does it have the same issue?

@crtrott I'm not certain, but I can pattern match the same change there as well

msimberg reviewed (approved):
Good catch, thanks. Looks good to me.

ndellingwood committed:
hpx: similar updates to constrain hpx_thread_buffer size

updates the setup() and execute_range() member routines of the ParallelReduce specialization for HPX TeamPolicy

Signed-off-by: Nathan Ellingwood <ndellin@sandia.gov>

ndellingwood commented:
@crtrott I applied similar changes in [6fa0ad0](https://github.com/kokkos/kokkos/pull/8147/commits/6fa0ad02f21f4d7891dc8231886a22fea9809a83) for the ParallelReduce specialization https://github.com/kokkos/kokkos/pull/8147/files#diff-76a7214a6cee6d4ac5794386cf6a2e6cd335022e3bf6b32275cf0106ce367de1R1595-R1598 , is this sufficient?

ndellingwood commented:
The [SNL-CI / SNL_HIP_ROCM_5_6_1 (ON)](https://github.com/kokkos/kokkos/actions/runs/15447155658/job/43479842972?pr=8147) failure is unrelated (it does not test HPX), but I cannot access the [continuous-integration/jenkins/pr-merge](https://cloud1.cees.ornl.gov/jenkins-ci/job/Kokkos/job/PR-8147/4/display/redirect) results

```
504 Gateway Time-out
```

dalg24 commented:
Please rebase now that we merged #8145 and increase the number of threads

ndellingwood committed:
snl-ci: remove num threads constraint on HPX test

Signed-off-by: Nathan Ellingwood <ndellin@sandia.gov>

ndellingwood commented:
> Please rebase now that we merged #8145 and increase the number of threads

@dalg24 done in [916fa1a](https://github.com/kokkos/kokkos/pull/8147/commits/916fa1a9ce56256f24b5e38cb0f0279d343a4bab) , I dropped the setting for KOKKOS_NUM_THREADS in that commit

dalg24 commented:
Make sure you add this to the 4.7 changelog
